src/Source/VectorTilesSource.js

```diff
@@ -86,13 +86,23 @@ class VectorTilesSource extends TMSSource {
                 } else if (ffilter(layer)) {
                     // TODO: add support for expressions
                     // https://docs.mapbox.com/mapbox-gl-js/style-spec/expressions
-                    const stops = [];
-                    stops.push(layer.minzoom == undefined ? 2 : layer.minzoom);
+                    let stops = [];
                     checkStopsValues(layer.layout, stops);
                     checkStopsValues(layer.paint, stops);
+
+                    let minStop = Math.min(...stops);
+                    // if none is found, default to 2
+                    minStop = (minStop == Infinity) ? 2 : minStop;
+                    // compare to layer.minzoom and take the highest
+                    minStop = (layer.minzoom == undefined) ? minStop : Math.max(layer.minzoom, minStop);
```

Comment:
`layer.minzoom` is never defined?

Reply:
It can be defined, and undefined too https://docs.mapbox.com/mapbox-gl-js/style-spec/layers/#minzoom

Reply:
Actually, the minimum display zoom option is `Source.zoom.min`,  but it should not represent the layer display. We should do a review and add this option also in `Layer`.

Reply:
Ok, I changed all `minzoom`/`maxzoom` to `zoom.min/zoom.max`.

Reply:
could you change `Layer#minzoom`/`maxzoom` to `Layer#zoom.min/zoom.max`?

Reply:
This has nothing to do with `Layer/Layer`, as it is a mapbox thing here (see link above). So I can't change that.

Reply:
sorry, I complete muddle itowns layer and mapbox layer.